packages/flutter/lib/src/painting/circle_border.dart

```diff
@@ -42,58 +58,56 @@ class CircleBorder extends OutlinedBorder {
   }
 
   @override
-  ShapeBorder scale(double t) => CircleBorder(side: side.scale(t));
+  ShapeBorder scale(double t) => CircleBorder(side: side.scale(t), eccentricity: eccentricity);
 
   @override
   ShapeBorder? lerpFrom(ShapeBorder? a, double t) {
     if (a is CircleBorder) {
-      return CircleBorder(side: BorderSide.lerp(a.side, side, t));
+      return CircleBorder(
+        side: BorderSide.lerp(a.side, side, t),
+        eccentricity: ui.lerpDouble(a.eccentricity, eccentricity, t)!,
```

Comment:
`t` can be > 1.0 or < 0.0, which will then fire the assert in the CircleBorder constructor. We should clamp the output of lerpDouble here.

Reply:
Oh, I never knew that! Fixed!